OptimizelySDK/Event/EventFactory.cs

```diff
@@ -192,19 +192,32 @@ ProjectConfig config
             }
 
             var attributesList = new List<VisitorAttribute>();
+            var invalidUserAttributeKeys = new List<string>();
 
             if (userAttributes != null)
             {
-                foreach (var validUserAttribute in userAttributes.Where(attribute =>
-                             Validator.IsUserAttributeValid(attribute)))
+                foreach (var userAttribute in userAttributes)
                 {
-                    var attributeId = config.GetAttributeId(validUserAttribute.Key);
-                    if (!string.IsNullOrEmpty(attributeId))
+                    if (Validator.IsUserAttributeValid(userAttribute))
+                    {
+                        var attributeId = config.GetAttributeId(userAttribute.Key);
+                        if (!string.IsNullOrEmpty(attributeId))
```

Comment:
What about switching 204 and 201, so we only log invalid type attributes for attribute ids defined in datafile.
We do not care about types of not-registered attributes.

Reply:
Let me know what you think @jaeopt